sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -547,7 +586,23 @@ export class EventHubReceiver extends LinkEntity {
         );
         // It is possible for someone to close the receiver and then start it again.
         // Thus make sure that the receiver is present in the client cache.
-        if (!this._context.receivers[this.name]) this._context.receivers[this.name] = this;
+        if (!this._context.receivers[this.name]) {
+          // Check if the receiver already exists and remove the existing reference from the cache.
```

Comment:
Can another solution be that we dont update `this.name` until the new receiver is created?
i.e Dont update this.name at https://github.com/Azure/azure-sdk-for-js/blob/33c23ffd716a1b6769744da1282c1bf4eed9b454/sdk/eventhub/event-hubs/src/eventHubReceiver.ts#L636
Update the `name` only in the receiver options.
Use this name from the receiver options for logging purposes between option creation and init()
Once receiver is created, run `deleteFromCache` , update `this.name` and then update the cache?